doc/source/whatsnew/v0.18.0.txt

```diff
@@ -275,6 +276,171 @@ other anchored offsets like ``MonthBegin`` and ``YearBegin``.
+
+   .. code-block:: python
+
+      In [4]: r = df.resample('2s')
+
+      In [6]: r*10
+      pandas/tseries/resample.py:80: FutureWarning: .resample() is now a deferred operation
+      use .resample(...).mean() instead of .resample(...)
+
+      Out[6]:
+                            A         B         C         D
+      2010-01-01 09:00:00  4.857476  4.473507  3.570960  7.936154
+      2010-01-01 09:00:02  8.208011  7.943173  3.640340  5.310957
+      2010-01-01 09:00:04  4.339846  3.145823  4.241039  6.257326
+      2010-01-01 09:00:06  6.249881  6.097384  6.331650  6.124518
+      2010-01-01 09:00:08  5.104699  5.343172  5.732009  8.069486
+
+   Assignment operations will raise a ``ValueError``:
+
+   .. code-block:: python
+
+      In [7]: r.iloc[0] = 5
+      ValueError: .resample() is now a deferred operation
+            use .resample(...).mean() instead of .resample(...)
+            assignment will have no effect as you are working on a copy
+
+**New API**:
+
+Now, you write ``.resample`` as a 2-stage operation like groupby, which
+yields a ``Resampler``.
+
+.. ipython:: python
+
+   r = df.resample('2s')
+   r
+
+Downsampling
+''''''''''''
+
+You can then use this object to perform similar operations.
+These are downsampling operations (going from a lower frequency to a higher one).
+
+.. ipython:: python
+
+   r.mean()
+
+.. ipython:: python
+
+   r.sum()
+
+Furthermore, resample now supports ``getitem`` operations to perform the resample on specific columns.
+
+.. ipython:: python
+
+   r[['A','C']].mean()
+
+and ``.aggregate`` type operations.
+
+.. ipython:: python
+
+   r.agg({'A' : 'mean', 'B' : 'sum'})
+
+These accessors can of course, be combined
+
+.. ipython:: python
+
+   r[['A','B']].agg(['mean','sum'])
+
+Upsampling
+''''''''''
+
+.. currentmodule:: pandas.tseries.resample
+
+Upsampling operations take you from a higher frequency to a lower frequency. These are now
+performed with the ``Resampler`` objects with :meth:`~Resampler.pad`,
+:meth:`~Resampler.ffill`, and :meth:`~Resampler.upsample` methods.
+
+.. ipython:: python
+
+   s = Series(np.arange(5,dtype='int64'),
+              index=date_range('2010-01-01', periods=5, freq='Q'))
+   s
+
+Previously
+
+.. code-block:: python
+
+   In [6]: s.resample('M', fill_method='ffill')
+   Out[6]:
+   2010-03-31    0
+   2010-04-30    0
+   2010-05-31    0
+   2010-06-30    1
+   2010-07-31    1
+   2010-08-31    1
+   2010-09-30    2
+   2010-10-31    2
+   2010-11-30    2
+   2010-12-31    3
+   2011-01-31    3
+   2011-02-28    3
+   2011-03-31    4
+   Freq: M, dtype: int64
+
+New API
+
+.. ipython:: python
+
+   s.resample('M').ffill()
+
+.. note::
+
+   In the new API, you can either downsample OR upsample. The prior implementation would allow you to pass an aggregator function (like ``mean``) even though you were upsampling, provide a bit of confusion.
```

Comment:
Do we need to mention you can still obtain the same result in two steps? Something like `ts.resample(..).mean().fillna(..)`